swift/StableDiffusion/pipeline/StableDiffusionXLPipeline.swift

```diff
@@ -213,7 +213,7 @@ public struct StableDiffusionXLPipeline: StableDiffusionPipelineProtocol {
 
             // Switch to refiner if specified
             if let refiner = unetRefiner, step == refinerStartStep {
-                self.unet.unloadResources()
+                unet.unloadResources()
```

Comment:
This is cool. But if `reduceMemory` is `false`, I think both the normal unet and the refiner would be loaded while iterating through the `unet`. I would suggest we always `prewarm` the refiner so it's unloaded until needed.